File: gfg/POTD/2025-02-07/solution.js
/**
 * @param {Node} root
 * @returns {number[]}
 */

/*
class Node{
    constructor(data){
        this.data = data;
        this.left = null;
        this.right = null;
    }
}
*/

class Solution {
	inOrder(root) {
		let result = []
		this.traverse(root, result)
		return result
	}

	traverse(node, result) {
		if (!node) return
		this.traverse(node.left, result)
		result.push(node.data)
		this.traverse(node.right, result)
	}
}

File: gfg/POTD/2025-02-07/solution.py

'''
# Node Class:
class Node:
    def init(self,val):
        self.data = val
        self.left = None
        self.right = None
'''

class Solution:
    def inOrder(self, root):
        result = []
        self.traverse(root, result)
        return result

    def traverse(self, node, result):
        if not node:
            return
        self.traverse(node.left, result)
        result.append(node.data)
        self.traverse(node.right, result)
